File: Day-10/armstrong_number.cpp
#include <iostream>
#include <cmath>
using namespace std;

int countDigit(int n) 
{
    int count = 0;
    while(n)
    {
        count++;
        n /= 10;
    }
    return count;
}

bool armStrong(int num, int digit)
{
    int n = num, ans = 0, rem;

    while(n)
    {
        rem = n % 10;
        n /= 10;
        ans += pow(rem, digit);
    }
    if( ans == num)
    {
        return 1;
        }
    else
    {
        return 0;
    }
}

int main()
{
    int num;
    cout << "Enter Number: ";
    cin >> num;

    int digit = countDigit(num);
    cout << "Number of digits: " << digit << endl;

   cout<<armStrong(num , digit)<<endl;
    return 0;
}

File: Easy/ArmStrongNumber.java
package basic_programs;

import java.util.Scanner;

public class ArmStrongNumber {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter Number: ");
        int number = sc.nextInt();

        int originalNumber = number;
        int digitCount = 0, remainder, sum = 0, temp = number;

        // Count digits
        while (temp > 0) {
            temp /= 10;
            digitCount++;
        }
        temp = number;
        while (temp > 0) {
            remainder = temp % 10;
            sum += Math.pow(remainder, digitCount);
            temp /= 10;
        }
        if (originalNumber == sum) {
            System.out.println(originalNumber + " is an Armstrong Number.");
        } else {
            System.out.println(originalNumber + " is NOT an Armstrong Number.");
        }
    }
}
